src/SFA.DAS.EmploymentCheck.Data/Repositories/ApiOptionsRepository.cs

```diff
@@ -0,0 +1,56 @@
+﻿using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Table;
+using SFA.DAS.EmploymentCheck.Data.Repositories.Interfaces;
+using SFA.DAS.EmploymentCheck.Infrastructure.Configuration;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EmploymentCheck.Data.Repositories
+{
+    public class ApiOptionsRepository : IApiOptionsRepository
+    {
+        private const string RowKey = "ApiRetryOptions";
+        private const string StorageTableName = "EmploymentCheckApiOptions";
+        private readonly ApiConnectionConfiguration _apiConnectionConfiguration;
+        private CloudTable _table;
+
+        public ApiOptionsRepository(ApiConnectionConfiguration options)
+        {
+            _apiConnectionConfiguration = options;
+        }
+
+        private void InitTableStorage()
+        {
+            var storageAccount = CloudStorageAccount.Parse(_apiConnectionConfiguration.StorageAccountConnectionString);
+            var tableClient = storageAccount.CreateCloudTableClient();
+            _table = tableClient.GetTableReference(StorageTableName);
+            _table.CreateIfNotExistsAsync().ConfigureAwait(false);
+        }
+
+        private CloudTable GetTable()
+        {
+            if (_table == null) InitTableStorage();
+            return _table;
+        }
+
```

Comment:
it would be good to add unit test coverage for this, i believe @David-B-Read has managed to work out how to do it in employer incentives functions repository.

Reply:
This already a unit test